src/ir/dot/dot_generator.cc

```diff
@@ -57,14 +57,14 @@ class DotGeneratorHelper : public IRTraversingVisitor<DotGeneratorHelper> {
   std::string GetNodeName(const Operation& operation) {
     const Block* block = operation.parent();
     auto id = ssa_names_.GetOrCreateID(operation);
-    return absl::StrFormat("%s_%d",
+    return absl::StrFormat("%s_%s",
                            block == nullptr ? "g" : GetNodeName(*block), id);
   }
 
   // Returns the dot node name for the given `block`.
   std::string GetNodeName(const Block& block) {
     auto id = ssa_names_.GetOrCreateID(block);
-    return absl::StrFormat("B%d", id);
+    return absl::StrFormat("B%s", id);
```

Comment:
We only added the prefix because the ids were numbers earlier. Also, to avoid issues in dot rendering, we will need to enclose it in quotes.

```suggestion
    return absl::StrFormat("\"%s\"", id);
```

The tests need to be updated as well.

Reply:
Done!